adapters/socketio_berkeley.c

```diff
@@ -659,6 +500,220 @@ static int set_target_network_interface(int socket, char* mac_address)
 }
 #endif //__APPLE__
 
+static int initiate_socket_connection(SOCKET_IO_INSTANCE* socket_io_instance)
```

Comment:
Here is the diff for reference: https://github.com/Azure/azure-c-shared-utility/compare/master...RLeclair:azure-c-shared-utility:raulleclair/ipv6-comp

We are now calling `set_target_network_interface` here because of the socket move so we needed to move this function down